test/e2e/pod_resource_scaling/pod_resource_scaling_test.go

```diff
@@ -0,0 +1,431 @@
+				"value": "200Mi",
+			},
+		}
+		_, err = kClient.PatchPod(initialPod, ns, patch, "resize")
+		Ω(err).NotTo(HaveOccurred())
+
+		// Wait for resource update to be reflected
+		err = utils.WaitForCondition(func() bool {
+			currentPod, err := kClient.GetPod(pod.Name, ns)
+			if err != nil {
+				return false
+			}
+			// Memory is in bytes, 200Mi = 200 * 1024 * 1024 bytes
+			return currentPod.Spec.Containers[0].Resources.Requests.Memory().Value() == int64(200*1024*1024)
+		}, 10*time.Second, 120*time.Second)
+		Ω(err).NotTo(HaveOccurred())
+
+		finalPod, err = kClient.GetPod(pod.Name, ns)
+		Ω(err).NotTo(HaveOccurred())
+		Ω(finalPod.Status.StartTime).To(Equal(initialStartTime), "Pod should not have restarted")
+		Ω(finalPod.Status.ContainerStatuses[0].RestartCount).To(Equal(initialRestartCount), "Container should not have restarted")
+		verifyYunikornResourceUsage(finalPod, "memory", 200*1024*1024)
+	})
+
+	ginkgo.It("Pod resources(cpu/memory) resize down", func() {
+		// Create pod with initial resources
+		pod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: podName,
+				Labels: map[string]string{
+					"app": podName,
+				},
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:  "nginx",
+						Image: "nginx:latest",
+						Resources: v1.ResourceRequirements{
+							Requests: v1.ResourceList{
+								"cpu":    resource.MustParse("200m"),
+								"memory": resource.MustParse("200Mi"),
+							},
+							Limits: v1.ResourceList{
+								"cpu":    resource.MustParse("200m"),
+								"memory": resource.MustParse("200Mi"),
+							},
+						},
+					},
+				},
+			},
+		}
+
+		// Create pod
+		_, err = kClient.CreatePod(pod, ns)
+		Ω(err).NotTo(HaveOccurred())
+
+		// Wait for pod running
+		err = kClient.WaitForPodRunning(ns, pod.Name, 60*time.Second)
+		Ω(err).NotTo(HaveOccurred())
+
+		// Check if pod is scheduled by YuniKorn and verify CPU allocation is 100m
+		verifyYunikornResourceUsage(pod, "vcore", 200)
+
+		// Get initial pod state
+		initialPod, err := kClient.GetPod(pod.Name, ns)
+		Ω(err).NotTo(HaveOccurred())
+		initialStartTime := initialPod.Status.StartTime
+		initialRestartCount := initialPod.Status.ContainerStatuses[0].RestartCount
+
+		patch := []map[string]interface{}{
+			{
+				"op":    "replace",
+				"path":  "/spec/containers/0/resources/limits/cpu",
+				"value": "100m",
+			},
+			{
+				"op":    "replace",
+				"path":  "/spec/containers/0/resources/requests/cpu",
+				"value": "100m",
+			},
+		}
+		_, err = kClient.PatchPod(initialPod, ns, patch, "resize")
+		Ω(err).NotTo(HaveOccurred())
+
+		// Wait for resource update to be reflected
+		err = utils.WaitForCondition(func() bool {
+			currentPod, err := kClient.GetPod(pod.Name, ns)
+			if err != nil {
+				return false
+			}
+			return currentPod.Spec.Containers[0].Resources.Requests.Cpu().MilliValue() == int64(100)
+		}, 10*time.Second, 120*time.Second)
+		Ω(err).NotTo(HaveOccurred())
```

Comment:
We have a wait loop inside `verifyYunikornResourceUsage()`, which is our main interest. This condition check can be dropped.